rfcs/0121-hardware-driver.md

```diff
@@ -0,0 +1,88 @@
+---
+feature: Use a more canonical name for hardware driver path and modules
+start-date: 2022-02-03
+author: @jonringer
+co-authors:
+shepherd-team: @edolstra @colemickens @Ericson2314
+shepherd-leader: @colemickens
+related-issues: https://github.com/NixOS/nixpkgs/issues/141803
+---
+
+# Summary
+[summary]: #summary
+
+Currently, NixOS mounts video drivers under the path `/run/opengl-driver/`,
+but should be mounted under a more generic `/run/current-system/drivers/` path.
```

Comment:
I'm not sure if this is actually going in the right direction. There's already the problem where nixpkgs binaries are hardcoded to the `/run/opengl-driver` path, which prevents them from working on non-NixOS systems by default. With this change, this becomes even more NixOS specific. NixGL was created to fix this problem, but that's only a workaround, not a permanent solution.

So I'd say rather than trying to fix the naming of something that's already problematic, how about we instead focus on fixing the problem at its core, then the misnaming will go away automatically when it's replaced with something better.

Reply:
"Don't let perfect be the enemy of good". 

I agree that the solution isn't ideal, but I can't think of a better one which can be achieved in a reasonable of time or significant work on how derivations work.

I just want Nixpkgs/NixOS in a better state. I'm fine delaying the heavier ramifications of using a different path (e.g `/run/current-system/sw/driver`), which have already been postponed for allowing for a more seamless upgrade path for users.

But I really just for the user facing "APIs" to be more intuitive (e.g. `hardware.gpu.drivers = [ "nvidia" ];` rather than `services.xserver.videoDrivers = [ "nvidia ];`). 

This RFC has already been open for ~10 months, it's almost impossible to get anything moved through in this RFC process; this really encourages people to just create a nixpkgs PR, and get a sympathetic committer to merge it and avoid the RFC process altogether.